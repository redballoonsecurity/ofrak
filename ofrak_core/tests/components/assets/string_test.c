
#include <stdio.h>


extern int longString(void);
extern int shortString(void);

// Generate bytes that look like a long ascii string (21 bytes) that will be matched as a string
// by the AsciiUnpacker. Assumes running on x86.
__asm__(".global longString\n\t"
    ".type longString, @function\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "push %r15\n\t"
    "and 0, %r15\n\t"
);

// Generate bytes that look like a short ascii string (7 bytes) that will not be matched as a string
// by the AsciiUnpacker because of the min length requirement. Assumes running on x86.
__asm__(".global shortString\n\t"
    ".type shortString, @function\n\t"
    "push %r15\n\t"
    "and 0, %r15\n\t"
);


int main() {
    printf("O");
    printf("h, hi");
    printf(" Marc!");
    printf("You are tearing me apart, Lisa!");
    return 0;
}
