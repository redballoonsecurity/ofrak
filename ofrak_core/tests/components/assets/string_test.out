version https://git-lfs.github.com/spec/v1
oid sha256:6362fe0a2954e363e28faead8351c4d65f65ffec4a62d21970064329233d09bc
size 16664
